### PR TITLE
Add timeout to extractor execution

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

To prevent a stuck or mis-configured connector from running forever.

### 🤓 What?

The original plan was to add a common `timeout` config option with a default value. However, it turns out that doing so [is difficult before Python 3.10](https://stackoverflow.com/questions/51575931/class-inheritance-in-python-3-7-dataclasses). Since this is more of a fail-safe mechanism than a configuration option, overriding it using an environment variable seems like a perfectly acceptable solution.

### 🧪 Tested?

```
% TIMEOUT=2 python -m metaphor.bigquery ~/Desktop/local/bigquery.yml
...
Traceback (most recent call last):
  File "/Users/marslan/.pyenv/versions/3.9.6/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/marslan/.pyenv/versions/3.9.6/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/marslan/Metaphor/connectors/metaphor/bigquery/__main__.py", line 6, in <module>
    cli_main("BigQuery metadata extractor", BigQueryExtractor)
  File "/Users/marslan/Metaphor/connectors/metaphor/common/cli.py", line 19, in cli_main
    extractor.run(config=config_cls.from_yaml_file(args.config))
  File "/Users/marslan/Metaphor/connectors/metaphor/common/extractor.py", line 52, in run
    raise TimeoutError(f"Process killed after {timeout} seconds")
TimeoutError: Process killed after 2.0 seconds
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "metaphor-connectors"
-version = "0.10.121"
+version = "0.10.123"
```
Comment (alyiwang) [inline pyproject.toml:3]: 0.10.122 ?

**Review by alyiwang — APPROVED** [review]
Otherwise, LGTM

File: pyproject.toml (hunk 1 above)
Comment (mars-lan) [reply pyproject.toml:3]: I was hoping https://github.com/MetaphorData/connectors/pull/301 will get in first :)

**Review by alyiwang — APPROVED** [review]
LGTM

**Pull request merged**